code/Player/PropPossession.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using Sandbox;
+using TTT.UI;
+
+namespace TTT;
+
+public partial class PropPossession : EntityComponent<Player>
+{
+	[Net] public Prop Prop { get; set; }
+	[Net, Local] public int Punches { get; set; }
+
+	// Clientside:
+	private PossessionNameplate _nameplate;
+	private PossessionInfo _info;
+
+	// Serverside:
+	private Player _player; // `Entity` property is cleared before OnDeactivate() called => need to store separately
+	private TimeUntil _timeUntilRecharge = 0;
+	private TimeUntil _timeUntilNextPunchAllowed = 0;
+
+	public PropPossession() { }
+
+	public PropPossession( Prop prop ) { Prop = prop; }
+
+	protected override void OnActivate()
+	{
+		_player = Entity;
+
+		if ( Host.IsServer )
+		{
+			_player.Camera = new FollowEntityCamera( Prop );
+			Prop.Owner = Entity;
+			_timeUntilRecharge = Game.PropPossessionRechargeTime;
+		}
+		else
+		{
+			if ( Local.Pawn is Player p && p.Status != PlayerStatus.Alive )
+				_nameplate = new PossessionNameplate( Entity, Prop );
+
+			if ( Entity.IsLocalPawn )
+				_info = new PossessionInfo( this ); // local pawn is possessing a prop
+		}
+	}
+
+	protected override void OnDeactivate() // `Entity` property is null when this is called
+	{
+		if ( Host.IsServer )
+		{
+			if ( _player.IsValid() && _player.Status != PlayerStatus.Alive )
+				_player.Camera = new FreeSpectateCamera();
+
+			if ( Prop.IsValid() )
+				Prop.Owner = null;
+		}
+		else
+		{
+			_nameplate?.Delete();
+			_info?.Delete();
+		}
+
+		_player = null;
+	}
+
+	[GameEvent.Player.StatusChanged]
+	private void OnStatusChanged( Player somePlayer, PlayerStatus oldStatus )
+	{
+		if ( somePlayer.Status == PlayerStatus.Alive )
+		{
+			// somePlayer has come alive
+			if ( Host.IsServer && somePlayer == Entity )
+			{
+				// _player is not possessing anything anymore since they have come alive
+				// we cannot immediately remove this component since that would remove all event handlers
+				// (including the one we are currently in) which would cause a InvalidOperationException
+				// (Collection was modified; enumeration operation may not execute)
+				RemoveThisComponentDelayed();
```

Comment:
This seems hacky, when I have some time I'll see if there is anything else we can do here.